varats-core/varats/project/project_util.py

```diff
@@ -274,36 +357,67 @@ def fetch(self) -> pb.LocalPath:
         Overrides ``Git`` s fetch to
           1. fetch the vara-test-repos repo
           2. extract the specified repo from the vara-test-repos repo
+          3. rename files that were made git_storable (e.g., .gitted) back to
+             their original name (e.g., .git)
 
         Returns:
             the path where the inner repo is extracted to
         """
+        self.__vara_test_repos_git.shallow = self.shallow
+        self.__vara_test_repos_git.clone = self.clone
+
         vara_test_repos_path = self.__vara_test_repos_git.fetch()
+        main_src_path = vara_test_repos_path / self.remote
+        main_tgt_path = local.path(target_prefix()) / self.local
 
-        # .gitted repo lies at vara_test_repos_path / self.remote
-        # check out as self.local
-        src_path = vara_test_repos_path / self.remote
-        tgt_path = local.path(target_prefix()) / self.local
+        # Extract main repository
+        if not os.path.isdir(main_tgt_path):
+            copy_renamed_git_to_dest(main_src_path, main_tgt_path)
 
-        mkdir("-p", tgt_path)
-        cp("-r", src_path + "/.", tgt_path)
+        return main_tgt_path
 
-        with local.cwd(tgt_path):
-            find(
-                ".", "-depth", "-name", ".gitted", "-execdir", "mv", "-i", "{}",
-                ".git", ";"
-            )
-            find(
-                ".", "-name", "gitmodules", "-execdir", "mv", "-i", "{}",
-                ".gitmodules", ";"
-            )
-            find(
-                ".", "-name", "gitattributes", "-execdir", "mv", "-i", "{}",
-                ".gitattributes", ";"
-            )
-            find(
-                ".", "-name", "gitignore", "-execdir", "mv", "-i", "{}",
-                ".gitignore", ";"
+    def version(self, target_dir: str, version: str = 'HEAD') -> pb.LocalPath:
+        """Overrides ``Git`` s version to create a new git worktree pointing to
+        the requested version."""
+
+        main_repo_src_local = self.fetch()
+        tgt_loc = pb.local.path(target_dir) / self.local
+        vara_test_repos_path = self.__vara_test_repos_git.fetch()
+        main_repo_src_remote = vara_test_repos_path / self.remote
+
+        mkdir('-p', tgt_loc)
+
+        # Extract main repository
+        cp("-r", main_repo_src_local + "/.", tgt_loc)
+
+        # Skip submodule extraction if none exist
+        if not Path(tgt_loc / ".gitmodules").exists():
+            return tgt_loc
```

Comment:
You still have to do the checkout, even if no submodules are present.

```suggestion
        if not Path(tgt_loc / ".gitmodules").exists():
            with pb.local.cwd(tgt_loc):
                git("checkout", "--detach", version)
            return tgt_loc
```